### PR TITLE
Add infrastructure to run Eventing E2E tests

### PR DESCRIPTION
**Pull request by marcobebway**
**Description**

Add infrastructure to run Eventing E2E tests.

### CODE REVIEW

File: prow/scripts/cluster-integration/kyma-integration-gardener-eventing.sh [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+  gardener::install_kyma
+fi
+
+# generate pod-security-policy list in json
+utils::save_psp_list "${ARTIFACTS}/kyma-psp.json"
+
+
+if [[ "${HIBERNATION_ENABLED}" == "true" ]]; then
+    gardener::hibernate_kyma
+    sleep 120
```
Comment (sayanh) [inline prow/scripts/cluster-integration/kyma-integration-gardener-eventing.sh:142]: How about making`sleep` configurable from job manifest?

File: prow/scripts/cluster-integration/kyma-integration-gardener-eventing.sh [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+    gardener::wake_up_kyma
+fi
+
+
+if [[ "${EXECUTION_PROFILE}" == "evaluation" ]] || [[ "${EXECUTION_PROFILE}" == "production" ]]; then
+    gardener::test_fast_integration_kyma
```
Comment (sayanh) [inline prow/scripts/cluster-integration/kyma-integration-gardener-eventing.sh:148]: Can we please add comments along the following lines:
- run fast integration tests (NATS based)
- switch to BEB and make sure eventing is ready
- run fast integration tests again (BEB based)

Bonus:
- switch to NATS again
- run fast integration tests again (NATS based)

Comment (marcobebway) [reply prow/scripts/cluster-integration/kyma-integration-gardener-eventing.sh:149]: This PR is about having the basic infra setup needed for the E2E tests.
In my opinion, those comments/scenarios should be well thought of when the real test is developed.

File: prow/scripts/cluster-integration/kyma-integration-gardener-eventing.sh (hunk 1 above)
Comment (marcobebway) [reply prow/scripts/cluster-integration/kyma-integration-gardener-eventing.sh:143]: Added a todo for that.

**Review by sayanh — APPROVED** [review]

**Review by sayanh — DISMISSED** [review]

**Review by sayanh — APPROVED** [review]

**Pull request merged**